x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/execution/sample/SampleIteratorTests.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.eql.execution.sample;
+
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static org.elasticsearch.xpack.eql.execution.sample.SampleIterator.matchSample;
+
+public class SampleIteratorTests extends ESTestCase {
+
+    public void testMatchSample() {
+        assertEquals(
+            asSearchHitsList(2, 1, 3),
+            matchSample(asList(asSearchHitsList(1, 1, 2), asSearchHitsList(1, 1, 1), asSearchHitsList(1, 1, 3)), 3)
```

Comment:
Can these search hits contain duplicates in practice?
It makes sense to test the algorithm in this case as well, just wondering if the algorithm can be simplified if it's not true (ie. avoid the backtracking by sorting the searchHits lists by size)

Reply:
I thought a bit about this, IMHO we should not test duplicates here. They are not allowed by construction (each query in the final reduce step should not produce duplicates), and they would also invalidate the general approach (ie. if you had duplicates, you could not rely on `size=n` and `terminate_after=n` in the final queries, without losing results).
The backtracking algorithm tolerates duplicate results, but IMHO it can be improved, and this test could fail with a different, yet valid, algorithm that does not tolerate duplicates.